hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormatVersion.java

```diff
@@ -80,18 +80,15 @@ public boolean hasHeader() {
 
   @Override
   public boolean hasFooter() {
-    switch (super.getVersion()) {
-      case DEFAULT_VERSION:
-        return false;
-      case 1:
-        return true;
-      default:
-        return false;
-    }
+    return hasLogFooterAndBlockLength();
```

Comment:
while I am in favor of cleanups, can we keep this PR to just removing BucketedIndex related changes alone? we can open a JIRA for these fixes.. wdyt?

Reply:
> while I am in favor of cleanups, can we keep this PR to just removing BucketedIndex related changes alone? we can open a JIRA for these fixes.. wdyt?

ok

Reply:
I am doing optimization when researching and learning the code. If this content is necessary, can I add another PR?

thanks